spanner/client.go

```diff
@@ -609,6 +609,9 @@ func (c *Client) Single() *ReadOnlyTransaction {
 	}
 	t.txReadOnly.qo.DirectedReadOptions = c.dro
 	t.txReadOnly.ro.DirectedReadOptions = c.dro
+	if c.ro.LockHint == sppb.ReadRequest_LOCK_HINT_EXCLUSIVE {
```

Comment:
Why do we need an if condition for unsetting? With this condition, there is one more lockhint called `LOCKHINT_SHARED`  and this will get applied to RO txns. Is that not a problem?

Can't we just do the below?
```
t.txReadOnly.ro.LockHint = sppb.ReadRequest_LOCK_HINT_UNSPECIFIED
```

Reply:
Done